docs: include a section with CircleCI instructions (#318)
game-ci/documentation: docs/11-circleci/02-activation.mdx

---
toc_max_heading_level: 4
---

# Activation

> The orb uses a Unity Editor instance to test and build your projects. These operations require the

Reviewer: Do all GameCI users use the Unity Editor? I was wondering if it would make sense to link out to the relevant page/docs here.

Author: Yup. Right now, GameCI is solely focused on Unity. If users get to this documentation, we can safely assume they know what the Editor is.

> instance to be activated before each job you run.

Reviewer 2: This part I just read in the previous document. Let's leave it away from this file to make it more concise.

Author: I'm repeating this here because the users won't always read the sections in order like you did. I plan to link the Activation section in a couple of error messages in the orb. And users coming from these messages will miss that informational context if I restrict it to the Getting Started section.

Let me know what you think.

Reviewer 2: Right. Or perhaps we should just leave it away from the getting started section.

I'm good with whatever you choose to do here.

Author: IIRC, I condensed the information on Getting Started based on another suggestion you made. I think we are good now.

Unity separates its licenses between `Personal` and `Professional`. The former applies to the _free_
and the latter to the _Pro_ and _Plus_ versions. Based on your use case proceed to the
[Personal](#personal-license) or [Professional](#professional-license) license section.

## Personal license

Follow these steps if you are using Unity's _free_ version.

> _**Note:** Do NOT follow the steps for the personal license if you have a professional license._

#### Mental model

> 1. Use the Unity orb to create an activation file.
> 2. Convert the activation file to a license file.
> 3. Encode the license file and save it in a CircleCI Context.

Reviewer 2: ```suggestion
1. Use the Unity Orb to create an activation file.
2. Use the activation file to retrieve a license file from the Unity website.
3. Encode the license file and save it in a CircleCI Context.
```

Author: Done ✔️

> #### Creating an activation file

Reviewer 2: ```suggestion
#### Acquiring an activation file
```

Author: Done ✔️

> Add the workflow below to a [project set up in CircleCI](https://circleci.com/docs/getting-started).
> As long as you can access it in the CircleCI Dashboard and download its artifacts, it can be any
> project.

Reviewer 2: ```suggestion
Add the following workflow to your [project set up in CircleCI](https://circleci.com/docs/getting-started). You need to be able to access the CircleCI Dashboard to download its artifacts.
```

> As long as you can access it in the CircleCI Dashboard and download its artifacts, it can be any

Reviewer: ```suggestion
As long as you can access it in the CircleCI web app and download its artifacts, it can be any
```

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  create-unity-activation-file:
    jobs:
      - unity/create-activation-file
```

> If you require an activation file for a specific Editor version, provide its semantic version. The
> available options can be found on the [versions](/docs/docker/versions) page.

Reviewer 2: ```suggestion
If you require an activation file for a specific version of the Unity Editor, you need to provide its version number as an argument. The available versions can be found on our [versions](/docs/docker/versions) page.
```

Author: Done ✔️

[versions](/docs/docker/versions) page.

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  create-unity-activation-file:
    jobs:
      - unity/create-activation-file:
          editor_version: 2022.1.11f1
```

> Commit and push the new workflow and navigate to your project in the CircleCI Dashboard. Identify

Reviewer: ```suggestion
Commit and push the new workflow and navigate to your project in the CircleCI web app. Identify
```

> Commit and push the new workflow and navigate to your project in the CircleCI Dashboard. Identify
> the pipeline triggered by your changes and open the `unity/create-activation-file` job. In the
> `ARTIFACTS` tab, you will encounter the `Unity.alf` file. Download it and follow the next section
> steps

Reviewer 2: ```suggestion
Commit and push the new workflow and navigate to your project in the CircleCI Dashboard. Identify
the pipeline triggered by your changes and open the `unity/create-activation-file` job. In the
`ARTIFACTS` tab, you will find the `Unity.alf` file. Download it. This is your activation file.
```

Author: Done ✔️

> #### Converting the activation file

Reviewer 2: ```suggestion
#### Acquiring the license file
```

Author: Done ✔️. But I'm also replacing `retrieving` with `acquiring` [here](https://github.com/game-ci/documentation/pull/318#discussion_r944739694).

> 1. Navigate to the [manual activation](https://license.unity3d.com/manual) page and upload the
>    `Unity.alf` file.

Reviewer 2: ```suggestion
1. Navigate to the [manual activation](https://license.unity3d.com/manual) page and upload the
   `Unity.alf` activation file.
```

Author: Done ✔️

2. Answer the prompted questions.

> 3. Download the license file (Unity_v20XX.x.ulf). It's ok if the numbers don't match your exact
>    Unity version.

Reviewer 2: ```suggestion
3. Download the license file (Unity_v20XX.x.ulf). 

_Note: It is fine if the version number in the file name does not match your exact Unity version._
```

Author: Done ✔️

#### Encoding the license

Encode the contents of your `Unity_v20XX.x.ulf` file to base64. If you have the `base64` command
line utility you can use:

```
cat Unity_v20XX.x.ulf | base64
```

> #### Preparing the Context

Reviewer 2: ```suggestion
#### Configuring your Context
```

Author: Done ✔️

> 1. Navigate to [CircleCI](https://app.circleci.com/) ⇢ `Organization Settings` ⇢ `Contexts` ⇢
>    `Create Context` and choose a context name. Alternatively, open an existing Context that you want
>    to use.

Reviewer 2: ```suggestion
1. Navigate to [CircleCI](https://app.circleci.com/) ⇢ `Organization Settings` ⇢ `Contexts` ⇢
   `Create Context` and choose a name for your context (e.g. `myProjectContext`). Alternatively, open an existing Context that you would like to use.
```

Author: Done ✔️

2. Create the following environment variables:
   - `UNITY_ENCODED_LICENSE` - _(Paste the content from the
     [Encoding the License](#encoding-the-license) section here)_
   - `UNITY_USERNAME` - _(Add the email address that you use to log in to Unity)_
   - `UNITY_PASSWORD` - _(Add the password that you use to log in to Unity)_

> 3. Remove the job you added to your `.circleci/config.yml` on the
>    [Creating an Activation File](#creating-an-activation-file) section as it's no longer necessary.

Reviewer 2: ```suggestion
3. The job that you add before to get your activation file is no longer necessary. You can safely remove or disable it.
```

Author: All action items in this list start with a verb: navigate, create and remove. Changing this pattern for one item will break the reading flow. However, I can't think of a better alternative, so I'll make the change if you are set on it.

Reviewer 2: Right. I am definitely not set on pushing for a change that doesn't make sense.

I was trying to get across that it is optional. I didn't think the link was necessary here reading it from the eyes of someone who has just implemented that.

Author: I agree the link is too much coming from the perspective of someone going through the docs. I'll reword what you suggested to fit the format of the list.

> _**Note:** When changing the Unity version, you may need to repeat the same process._

## Professional license

Follow these steps if you are using Unity's _Pro_ or _Plus_ version.

> _**Note:** Do NOT follow the steps for the professional license if you have a personal license._

> 1. Subscribe to Unity Pro or Unity Plus, and get your Serial Key from the
>    [subscriptions page](https://id.unity.com/en/subscriptions).

Reviewer 2: ```suggestion
1. Get your Serial Key from the
   [subscriptions page](https://id.unity.com/en/subscriptions).
```

Author: Done ✔️

> 2. Navigate to [CircleCI](https://app.circleci.com/) ⇢ `Organization Settings` ⇢ `Contexts` ⇢
>    `Create Context` and choose a context name. Alternatively, open an existing Context that you want
>    to use.

Reviewer 2: ```suggestion
2. Navigate to [CircleCI](https://app.circleci.com/) ⇢ `Organization Settings` ⇢ `Contexts` ⇢
   `Create Context` and choose a name for your context (e.g. `myProjectContext`). Alternatively, open an existing Context that you would like to use.
```

Author: Done ✔️

3. Create the following environment variables:

>    - `UNITY_SERIAL` - _(Paste the content from **step 1** here. It should look like this:

Reviewer 2: ```suggestion
   - `UNITY_SERIAL` - _(Paste the serial that you acquired in step 1. It should look like this:
```

Author: Done ✔️

and the latter to the _Pro_ and _Plus_ versions. Based on your use case proceed to the
[Personal](#personal-license) or [Professional](#professional-license) license section.

> _**Note:** GameCI does not acquire nor store your Unity email, password or serial. They are
> required for activating the license during the build and test steps._

> 

Reviewer 2: I'd argue that modern tools would never do this. I don't think this note is really needed.

Author: I don't mind removing it. It is a copy-paste from the builder: https://github.com/game-ci/documentation/blob/main/docs/03-github/04-builder.mdx?plain=1#L26.

1. Subscribe to Unity Pro or Unity Plus, and get your Serial Key from the
   [subscriptions page](https://id.unity.com/en/subscriptions).
2. Navigate to [CircleCI](https://app.circleci.com/) ⇢ `Organization Settings` ⇢ `Contexts` ⇢
   `Create Context` and choose a context name. Alternatively, open an existing Context that you want
   to use.
3. Create the following environment variables:
   - `UNITY_SERIAL` - _(Paste the content from **step 1** here. It should look like this:
     `XX-XXXX-XXXX-XXXX-XXXX-XXXX`)_
   - `UNITY_USERNAME` - _(Add the email address that you use to log in to Unity)_
   - `UNITY_PASSWORD` - _(Add the password that you use to log in to Unity)_

## Next steps

> With your context set, you are good to start testing and [building](/docs/circleci/build) your Unity

Reviewer 2: ```suggestion
Now you are prepared to start testing and [building](/docs/circleci/build) your Unity
```

Author: Love it, thanks ❤️
